internal/resourceproviders/registration.go

```diff
@@ -22,6 +22,10 @@ import (
 // register them if it appears they are not. Note that this may fail if a resource provider is not available in the
 // current cloud environment (a warning message will be logged to indicate when a resource provider is not listed).
 func EnsureRegistered(ctx context.Context, client *providers.ProvidersClient, subscriptionId commonids.SubscriptionId, requiredRPs ResourceProviders) error {
+	if len(requiredRPs) == 0 {
+		log.Printf("[DEBUG] No Resource Provider to register")
+		return nil
+	}
```

Comment:
Initially I had the same thought as @tombuildsstuff that we should continue to populate the cache to provide full validation for the `azurerm_resource_provider_registration` resource. But I also think it would be nice to offer a way to disable the cache, both for speed, and for cases where new RPs are not yet published (which has happened before).

I think that gating this behind a feature would be a good compromise - and we have a related one for enhanced validation that might be good to use already. WYDT?

```suggestion
	// Cache supported resource providers if RP registration and enhanced validation are not both disabled
	if len(requiredRPs) == 0 && !features.EnhancedValidationEnabled() {
		log.Printf("[DEBUG] Skipping populating the resource provider cache, since resource provider registration and enhanced validation are both disabled")
		return nil
	}

```